contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,572 @@
+name: local-raw-pv
+spec:
+  volumeMode: Block #proposed API change
+  capacity:
+    storage: 100Gi
+  local:
+    path: /dev/xvdc #device path
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete 
+```
+## Pod API Changes:
+To provide better specificy and ensure support of inline volumes, the following changes are proposed in the pod specification.
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-db
+spec:
+    containers
+    - name: mysql
+      image: mysql
+      volumeDevices: #proposed API change
+      - name: my-db-data
+        devicePath: /dev/xvda #proposed API change
+    volumes:
+    - name: my-db-data
+      persistentVolumeClaim:
+	claimName: raw-pvc
+```
+## Storage Class non-API Changes:
+For dynamic provisioning, it is assumed that values pass in the parameter section are opaque, thus the introduction of utilizing
+fsType in the StorageClass can be used by the provisioner to indicate how to create the volume. The proposal for this value is
+defined here:
+https://github.com/kubernetes/kubernetes/pull/45345 
+Therefore, a provisioner could potentially provision a block device and install the filesystem onto it by indicating the volumeMode
+as 'block' but the fsType as 'xfs'.
+This section is provided as a general guideline, but each provisioner may implement their parameters independent of what is defined
+here. It is our recommendation that the volumeMode in the PVC be the guidance for the provisioner and overrides the value given in the fstype. Therefore a provisioner should be able to ignore the fstype and provision a block device if that is what the user requested via the PVC and the provisioner can support this.
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: block-volume
+provisioner: kubernetes.io/local-block-glusterfs
+parameters:
+```
+The provisioner (if applicable) should validate the parameters and return and error if the combination specified is not supported.
+This also allows the use case for leveraging a Storage Class for utilizing pre-defined static volumes. By labeling the Persistent Volumes
+with the Storage Class, volumes can be grouped and used according to how they are defined in the class.
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: block-volume
+provisioner: no-provisioning 
+parameters:
+```
+
+# Pod Security Policy (PSP) Changes:
+Since the utilization of block devices can pose a risk to possible kernel manipulation by malicious users, it may be desirable for the administrator to restrict the usage entirely within a cluster. A similar convention is used with hostPath in that for some kubernetes 
+implmentations it is disabled by default.
+Thus, the PSP can define whether a validating pod can request the usage of such devices through the volumeMode label in the PV/PVC.
+The changes to the pod security policy for volumes would include a 'localRawBlockVolumes' and 'newtworkRawBlockVolumes' parameters as such:
```

Comment:
all the existing volume parameters map to possible volume types in the pod spec

Reply:
@liggit thanks for catching we have had some changes to the API to add volumeDevices and thus we will validate against this in the pod